platform/mac/do_mac_bundle.sh

```diff
@@ -34,6 +48,48 @@ cat <<EOF >"${INSTALL_DIR}/bundle/Contents/Info.plist"
         <string>APPL</string>
         <key>CFBundleIconFile</key>
         <string>icon</string>
+        <key>CFBundleDocumentTypes</key>
+        <array>
+            <dict>
+                <key>CFBundleTypeExtensions</key>
+                <array>
+                    <string>azw</string>
+                    <string>cbz</string>
+                    <string>chm</string>
+                    <string>djv</string>
+                    <string>djvu</string>
+                    <string>doc</string>
+                    <string>docx</string>
+                    <string>epub</string>
+                    <string>fb2</string>
+                    <string>htm</string>
+                    <string>html</string>
+                    <string>md</string>
+                    <string>mobi</string>
+                    <string>pdb</string>
+                    <string>pdf</string>
+                    <string>prc</string>
+                    <string>rtf</string>
+                    <string>txt</string>
+                    <string>xhtml</string>
+                    <string>xps</string>
+                </array>
+                <key>CFBundleTypeIconFile</key>
+                <string>icon</string>
+                <key>CFBundleTypeName</key>
+                <string>docs</string>
+                <key>CFBundleTypeRole</key>
+                <string>Viewer</string>
+            </dict>
+	</array>
```

Comment:
Not sure if it matters, but `CFBundleTypeExtensions` is "Deprecated in OS X v10.5."(see [documentation](https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/CoreFoundationKeys.html#//apple_ref/doc/uid/20001431-101685)). Also the outer `</array>` closing tag should be indented once more.

Reply:
They still work in Catalina. Apple guidelines now recommend using UTIs. OS supported ones are in https://developer.apple.com/library/archive/documentation/Miscellaneous/Reference/UTIRef/Articles/System-DeclaredUniformTypeIdentifiers.html#//apple_ref/doc/uid/TP40009259-SW1

I, personally, found too much of work trying to figure out which long name matches each extension (and adding our own UTI's, since most of the types are not recognized by osx), so I opted for document extensions and did a quick copy from https://github.com/koreader/koreader/blob/master/plugins/calibre.koplugin/extensions.lua :octocat: 

But yeah. UTIs are welcome!

Reply:
Well, if it works, then it's alright. Please check indentation, I believe the last `</array>` here uses [Tab], while the rest use spaces.
![image](https://user-images.githubusercontent.com/10298730/87251884-28043600-c46f-11ea-9592-e3b6fd43ee9a.png)

Reply:
> Well, if it works, then it's alright.

Yeah. Thanks for pointing it out BTW. I did check the last version of calibre (4.20) and they are still using `CFBundleTypeExtensions`. As calibre has hundred of thousand of macOS users I think it is safe to use deprecated methods until they fail. Then copy the new UTIs from calibre as it supports all the filetypes we're supporting and much more!.